std/haxe/macro/Compiler.hx

```diff
@@ -577,3 +615,20 @@ enum abstract NullSafetyMode(String) to String {
 	**/
 	var StrictThreaded;
 }
+
+typedef MetadataDescription = {
```

Comment:
Could we add some documentation? The difference between targets and platforms needs clarification.

Reply:
That TODO somehow escaped from my list :D
I also want to use proper enums (currently available in `haxe.display.Display` as `MetadataTarget` and `Platform`) instead of string. Will try to update that tomorrow

Reply:
I added _very_ basic documentation and used enum abstracts mentioned above. Is more needed?